tests/foreman/api/test_hostgroup.py

```diff
@@ -152,6 +163,826 @@ def test_verify_bugzilla_1107708(self):
         self.assertEqual(len(host_attrs['all_puppetclasses']), 1)
         self.assertEqual(host_attrs['all_puppetclasses'][0]['name'], 'ntp')
 
+    @tier1
+    def test_positive_create_with_name(self):
+        """Create a hostgroup with different names
+
+        @feature: HostGroup
+
+        @assert: A hostgroup is created with expected name
+        """
+        for name in valid_hostgroups_list():
+            with self.subTest(name):
+                hostgroup = entities.HostGroup(
+                    location=[self.loc],
+                    name=name,
+                    organization=[self.org],
+                ).create()
+                self.assertEqual(name, hostgroup.name)
+
+    @tier2
+    def test_positive_create_with_parent(self):
+        """Create a hostgroup with parent hostgroup specified
+
+        @feature: HostGroup
+
+        @assert: A hostgroup is created with expected parent hostgroup assigned
+        """
+        parent_hostgroup = entities.HostGroup(
+            location=[self.loc],
+            organization=[self.org],
+        ).create()
+        hostgroup = entities.HostGroup(
+            location=[self.loc],
+            organization=[self.org],
+            parent=parent_hostgroup,
+        ).create()
+        self.assertEqual(hostgroup.parent.read().name, parent_hostgroup.name)
+
+    @tier2
+    def test_positive_create_with_env(self):
+        """Create a hostgroup with environment specified
+
+        @feature: HostGroup
+
+        @assert: A hostgroup is created with expected environment assigned
+        """
+        env = entities.Environment(
+            location=[self.loc],
+            organization=[self.org],
+        ).create()
+        hostgroup = entities.HostGroup(
+            environment=env,
+            location=[self.loc],
+            organization=[self.org],
+        ).create()
+        self.assertEqual(hostgroup.environment.read().name, env.name)
+
+    @tier2
+    def test_positive_create_with_os(self):
+        """Create a hostgroup with operating system specified
+
+        @feature: HostGroup
+
+        @assert: A hostgroup is created with expected operating system assigned
+        """
+        arch = entities.Architecture().create()
```

Comment:
Seems that this test can be simplified to:

```python
        os = entities.OperatingSystem().create()
        hostgroup = entities.HostGroup(
            architecture=os.architecture,
            location=[self.loc],
            operatingsystem=os,
            organization=[self.org],
            ptable=os.ptable,
        ).create()
        self.assertEqual(hostgroup.operatingsystem.read().name, os.name)
```

I'm not sure if the OS create is going to create those dependent entities. This is a guess since I've been a long time away from writing API tests.